File: EASY/TEMPERATURES/python3/temperatures.py
# https://www.codingame.com/ide/puzzle/temperatures
n = int(input())  # the number of temperatures to analyse
temps = input()  # the n temperatures expressed as integers ranging from -273 to 5526
if n !=0:
    array = temps.split(' ')
    min = int(array[0])
    for i in range(1, n):
        if (abs(min)>abs(int(array[i]))  or (abs(min)==abs(int(array[i])) and int(array[i])>min)):
            min = int(array[i])
else:
    min = 0
print(str(min))

File: EASY/TEMPERATURES/php/temperatures.php
<?php
fscanf(STDIN, "%d",
    $n // the number of temperatures to analyse
);
$temps = stream_get_line(STDIN, 256 + 1, "\n"); // the n temperatures expressed as integers ranging from -273 to 5526
if($n !=0){
    $array = explode(' ', $temps);
    $min = (int)$array[0];
    foreach($array as $item){
        $item = (int)$item;
        if (abs($min)>abs($item) or (abs($min)==abs($item) and $item>$min))
            $min = $item;
    }
}
else
    $min = 0;
echo("$min\n");
